Jenkinsfile

```diff
@@ -1,48 +1,67 @@
 pipeline {
-    agent any 
+    agent any
     stages {
-	stage('Install dependencies and download data') {
-	    steps {
-		echo "Dependencies"
-		sh """
-        source /var/lib/jenkins/py3env/bin/activate
-        env
 
-        cd ${env.WORKSPACE}
-        pip install -r requirements.txt
-
-        echo ${env.JOB_NAME}
-        rm -rf /tmp/${env.JOB_NAME}
-        mkdir -p /tmp/${env.JOB_NAME}/sample0
-        cd /tmp/${env.JOB_NAME}
-        git clone https://github.com/ucgmsim/qcore.git
-
-        ln -s $HOME/data/testing/slurm_gm_workflow/SGMW /tmp/${env.JOB_NAME}/build
-        cd /tmp/${env.JOB_NAME}/sample0
-        cp -r $HOME/data/testing/slurm_gm_workflow/PangopangoF29_HYP01-10_S1244/* .
+        stage('Settin up env') {
+            steps {
+                echo "[[ Start virtual environment ]]"
+                sh """
+                    echo "[ Current directory ] : " `pwd`
+                    echo "[ Environment Variables ] "
+                    env
+# Each stage needs custom setting done again. By default /bin/python is used.
+                    source /var/lib/jenkins/py3env/bin/activate
+                    mkdir -p /tmp/${env.JOB_NAME}/${env.ghprbActualCommit}
+# I don't know how to create a variable within Jenkinsfile (please let me know)
```

Comment:
are these comments suppose to be here? 😂 
if so, try these two method:
1. define it as a globle before the pipeline
using `def virtenv = /tmp/${env.JOB_NAME}/${env.ghprbActualCommit}/venv`
2. inside script section:
`script {
    virtenv = /tmp/${env.JOB_NAME}/${env.ghprbActualCommit}/venv
}`
using #2 requires  env.virtenv  if its going to be used in different steps.

Reply:
Thanks - I'll give a go

Reply:
Ended up with environment directive - It only can keep one variable despite the documentation. Jenkins is such a joy. 
http://13.238.107.244:8080/job/slurm_gm_workflow/345/console